src/java/org/apache/cassandra/db/ClusteringComparator.java

```diff
@@ -353,6 +353,8 @@ public <V> Clustering<V> clusteringFromByteComparable(ValueAccessor<V> accessor,
     {
         ByteComparable.Version version = ByteComparable.Version.OSS42;
         ByteSource.Peekable orderedBytes = ByteSource.peekable(comparable.asComparableBytes(version));
+        if (orderedBytes == null)
+            return null;
```

Comment:
When would we actually get `null` here? Would it be better to just check for that case before the actual call to `clusteringFromByteComparable()` to avoid making a change here, when existing callers might assume `orderedBytes != null`.

Reply:
I know there aren't any existing callers outside SAI in this branch, but I assume there are others (trie-indexed SSTables?)